enhancements/machine-config/on-cluster-builds.md

```diff
@@ -0,0 +1,1083 @@
+---
+title: on-cluster-builds
+authors:
+  - cheesesashimi
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - mrunalp
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - mrunalp
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - joelspeed
+creation-date: 2023-11-06
+last-updated: 2023-11-06
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/MCO-834
+see-also:
+  - "/enhancements/ocp-coreos-layering.md"
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# On-Cluster Builds
+
+## Summary
+
+Change the Machine Config Operator (MCO) that is shipped in OpenShift to
+perform an on-cluster build (OCB) to combine the base OS image shipped
+with OpenShift, any admin-provided custom content, and the MachineConfig
+content managed by the MCO. This image will then be deployed to all of
+the cluster nodes using rpm-ostree (and later, bootc) to facilitate a
+singular, transactional, update.
```

Comment:
This mechanism sounds like it will be incompatible with the work that's ongoing to allow admins to inhibit reboots, has that been considered at all?

Reply:
You're correct to call this out. However, not every change needs to go into the container image build. We have some guidance in the upstream docs here https://docs.fedoraproject.org/en-US/bootc/dynamic-reconfiguration/

Reply:
It would be useful perhaps to have a workflow/example that demonstrates how the two would fit together, one where reboots are required and one where they are not perhaps?